Sliver panics after failed Wireguard connection

**Describe the bug**
Sliver panics after trying to connect to a server on port 53.
This panic might be caused by provider level manipulation of wg traffic on port 53. 
I verified that `nc -u host 53` traffic comes through on port 53. Wireguard failes to connect however. 

**To Reproduce**
Steps to reproduce the behavior:
1. Connect with a config like --wg host:53,host:123,host:161
2. Sliver connects only to 53 and panics after several retries with the following error message:
```
wireguard.go:174: Unable to connect to wg key exchange listener: connect tcp 100.64.0.1:1337: operation timed out
panic: ParseAddr(""): unable to parse IP
```

Also interesting is the miscounting on the connection tries. I observed this behavior several times already. Sometimes it jumps between 2 and 14 or other numbers randomly.

**Expected behavior**
Sliver should switch to the next Wireguard port after several failed retries (maybe even force switch after 60 seconds of failed attempts).
Catching the exception might help to restore normal behavior.

**Desktop (please complete the following information):**
 - OS: Debian/Windows
 - Version Server v1.5.29 - 32e4ed63e98ada7ca4283e64d1fb0b7ee60322fc

**Additional context**
Connection log
```
2022/10/06 14:21:03 wireguard.go:263: Successfully set wg device config
DEBUG: [c2/wg] 2022/10/06 14:21:03 peer(w4sk…BiiQ) - Routine: sequential sender - started
DEBUG: [c2/wg] 2022/10/06 14:21:03 peer(w4sk…BiiQ) - Routine: sequential receiver - started
DEBUG: [c2/wg] 2022/10/06 14:21:03 Routine: receive incoming v4 - started
DEBUG: [c2/wg] 2022/10/06 14:21:03 Routine: receive incoming v6 - started
DEBUG: [c2/wg] 2022/10/06 14:21:03 UDP bind has been updated
DEBUG: [c2/wg] 2022/10/06 14:21:03 Interface state was Down, requested Up, now Up
2022/10/06 14:21:03 wireguard.go:168: Initial wg connection. Attempting to connect to wg key exchange listener
DEBUG: [c2/wg] 2022/10/06 14:21:03 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:08 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [c2/wg] 2022/10/06 14:21:08 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:13 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [c2/wg] 2022/10/06 14:21:13 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:18 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [c2/wg] 2022/10/06 14:21:18 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:24 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 3)
DEBUG: [c2/wg] 2022/10/06 14:21:24 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:29 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 4)
DEBUG: [c2/wg] 2022/10/06 14:21:29 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:34 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 5)
DEBUG: [c2/wg] 2022/10/06 14:21:34 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:39 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [c2/wg] 2022/10/06 14:21:39 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:44 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 3)
DEBUG: [c2/wg] 2022/10/06 14:21:44 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:49 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 4)
DEBUG: [c2/wg] 2022/10/06 14:21:49 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:21:55 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 5)
DEBUG: [c2/wg] 2022/10/06 14:21:55 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:00 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 6)
DEBUG: [c2/wg] 2022/10/06 14:22:00 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:05 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 7)
DEBUG: [c2/wg] 2022/10/06 14:22:05 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:10 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [c2/wg] 2022/10/06 14:22:10 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:15 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 3)
DEBUG: [c2/wg] 2022/10/06 14:22:15 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:20 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 4)
DEBUG: [c2/wg] 2022/10/06 14:22:20 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:25 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 5)
DEBUG: [c2/wg] 2022/10/06 14:22:25 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:30 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 6)
DEBUG: [c2/wg] 2022/10/06 14:22:30 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:36 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 7)
DEBUG: [c2/wg] 2022/10/06 14:22:36 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:41 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 8)
DEBUG: [c2/wg] 2022/10/06 14:22:41 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:46 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 9)
DEBUG: [c2/wg] 2022/10/06 14:22:46 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:51 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 10)
DEBUG: [c2/wg] 2022/10/06 14:22:51 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:22:56 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 11)
DEBUG: [c2/wg] 2022/10/06 14:22:56 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:23:01 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 12)
DEBUG: [c2/wg] 2022/10/06 14:23:01 peer(w4sk…BiiQ) - Sending handshake initiation
DEBUG: [c2/wg] 2022/10/06 14:23:07 peer(w4sk…BiiQ) - Handshake did not complete after 5 seconds, retrying (try 13)
DEBUG: [c2/wg] 2022/10/06 14:23:07 peer(w4sk…BiiQ) - Sending handshake initiation
2022/10/06 14:23:10 wireguard.go:174: Unable to connect to wg key exchange listener: connect tcp 100.64.0.1:1337: operation timed out
panic: ParseAddr(""): unable to parse IP

goroutine 1 [running]:
net/netip.MustParseAddr({0x0, 0x0})
        .../.sliver/go/src/net/netip/netip.go:141 +0x145
github.com/bishopfox/sliver/implant/sliver/transports/wireguard.bringUpWGInterface({0xc00002f420, 0xb}, 0x35, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0})
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/implant/sliver/transports/wireguard/wireguard.go:231 +0xaa
github.com/bishopfox/sliver/implant/sliver/transports/wireguard.WGConnect({0xc00002f420, 0xb}, 0x35)
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/implant/sliver/transports/wireguard/wireguard.go:204 +0x111
github.com/bishopfox/sliver/implant/sliver/transports.wgConnect.func3()
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/implant/sliver/transports/session.go:201 +0x459
main.sessionMainLoop(0xc000218000)
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/sliver.go:147 +0x5d
main.sessionStartup()
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/sliver.go:120 +0x148
main.main()
        ...FAMOUS_TELEPHONE/src/github.com/bishopfox/sliver/sliver.go:101 +0xc5
```

--- a/implant/sliver/transports/wireguard/wireguard.go
+++ b/implant/sliver/transports/wireguard/wireguard.go
@@ -227,6 +227,11 @@ func WGConnect(address string, port uint16) (net.Conn, *device.Device, error) {
 // bringUpWGInterface - First creates an inet.af network stack.
 // then creates a Wireguard device/interface and applies configuration
 func bringUpWGInterface(address string, port uint16, implantPrivKey string, serverPubKey string, netstackTunIP string) (tun.Device, *device.Device, *netstack.Net, error) {
+	if netstackTunIP == "" {
+		err := errors.New("[wireguard] Cannot connect to empty IP address")
+		return nil, nil, nil, err
+	}
+
 	tun, tNet, err := netstack.CreateNetTUN(
 		[]netip.Addr{netip.MustParseAddr(netstackTunIP)},
 		[]netip.Addr{netip.MustParseAddr("127.0.0.1")}, // We don't use DNS in the WG implant. Yet.

